extendr-api/src/deserializer.rs

```diff
@@ -725,21 +725,19 @@ impl<'de> Visitor<'de> for RobjVisitor {
     where
         E: serde::de::Error,
     {
-        if v > i32::MIN as i64 && v <= i32::MAX as i64 {
-            Ok((v as i32).into())
-        } else {
-            Ok((v as f64).into())
+        match i32::try_from(v) {
+            Ok(x) => x,
+            Err(_) => f64::try_from(v)?
```

Comment:
serialize section must be checked.

Reply:
Great catch. Can you make this a separate issue?

Reply:
@CGMossa ready, I updated the code, now only will replace the "as" but will keep the logic on the code, there is a new issue for that things: https://github.com/extendr/extendr/issues/732